test/e2e/lib/gutenberg/gutenberg-editor-component.js

```diff
@@ -65,11 +65,17 @@ export default class GutenbergEditorComponent extends AsyncBaseContainer {
 
 	async publish( { visit = false, closePanel = true } = {} ) {
 		await driverHelper.clickWhenClickable( this.driver, this.prePublishButtonSelector );
+
+		// Clicking publish too soon often fails, I suspect this is due to the slide out animation.
+		// A short sleep is enough to prevent these failures
+		await this.driver.sleep( 10 );
```

Comment:
Could you link to a job where this is failing, please? It would help ensure we're fixing the right issue here.
As for the fix - 10 milliseconds sound risky to me, and ideally, if the animation is indeed the issue here, we should find out the actual duration of it and use that value for the sleep.

Reply:
I'll see if I can trigger it on this one. Was just testing locally and seeing this problem 30% ~ of the time.

Reply:
The 10ms - I suspect its something like a next tick timing issue. 1ms wasn't enough but 10ms was, I could try something in-between. I don't know if the transition itself was the problem. I'm thinking a react re/render or something like that.

Reply:
[Similar to](https://github.com/Automattic/wp-calypso/pull/52090#discussion_r617143564) here is the failure, happened on my third run:
```
e2e trunk NODE_CONFIG_ENV=decrypted BROWSERSIZE=desktop ./node_modules/.bin/mocha specs/wp-likes-spec.js


  [WPCOM] Likes: (desktop)
    Like posts and comments @parallel
Logging in as e2eflowtestinggutenbergsimple
Publish panel already closed
      1) Login, create a new post and view it
FAILED: Taking screenshot of: 'https://wordpress.com/post/e2eflowtestinggutenbergsimple.wordpress.co/m'
      - Like post
      - Unlike post
      - Post comment
      - Like comment
      - Unlike comment
      - Like post as logged out user


  0 passing (1m)
  6 pending
  1 failing

  1) [WPCOM] Likes: (desktop)
       Like posts and comments @parallel
         Login, create a new post and view it:
     TimeoutError: Waiting for the element to become located and visible By(css selector, .components-snackbar)
Wait timed out after 22043ms
      at /home/louis/wp/wp-calypso/node_modules/selenium-webdriver/lib/webdriver.js:842:17
      at processTicksAndRejections (internal/process/task_queues.js:93:5)



Accepting an open alert: 'UnexpectedAlertOpenError: unexpected alert open: {Alert text : }
  (Session info: chrome=90.0.4430.72)'
(node:167621) UnhandledPromiseRejectionWarning: NoSuchSessionError: This driver instance does not have a valid session ID (did you call WebDriver.quit()?) and may no longer be used.
    at /home/louis/wp/wp-calypso/node_modules/selenium-webdriver/lib/webdriver.js:730:38
    at Object.thenFinally [as finally] (/home/louis/wp/wp-calypso/node_modules/selenium-webdriver/lib/promise.js:124:12)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
(Use `node --trace-warnings ...` to show where the warning was created)
(node:167621) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 2)
```

The test run loads a little slow, then when it enters the page title / text you see a quick flash where the sidebar should extend but then nothing and you wait for the timeout.

Adding the spinner displayed check you see the failure far sooner (you don't have to wait for the 60second publish timeout) but the cause is the same - the publish sidebar is never shown. (this is why I wanted to remove the 60 second timeout, seems like this issue would create a lot of false positives for an outlier publish time) 

Or at least, it is but publish is clicked too fast and the sidebar dismisses before the publish can actually happen.

It looks like the publish button closes the sidebar regardless but maybe the publish event isn't wired up until the sidebar is rendered (early maybe?) - I'll take a look over at Gutenberg and see if anything can explain this but I think its worth just dialing in a lower ms sleep and calling it solved.

Reply:
Looks like there is a few things that could lead to this sort of thing.

The [actions panel](https://github.com/WordPress/gutenberg/blob/4c81e6e215bf81779920ea7110ca4cacc8bea284/packages/edit-post/src/components/layout/actions-panel.js#L56) loads a [`PostPublishPanel`](https://github.com/WordPress/gutenberg/blob/3da717b8d0ac7d7821fc6d0475695ccf3ae2829f/packages/editor/src/components/post-publish-panel/index.js#L29). The actons panel has [hints at some unmounting logic](https://github.com/WordPress/gutenberg/blob/4c81e6e215bf81779920ea7110ca4cacc8bea284/packages/edit-post/src/components/layout/actions-panel.js#L51) that could be the problem.

But it could also be something to do with the way we get [`hasPublishAction`](https://github.com/WordPress/gutenberg/blob/3da717b8d0ac7d7821fc6d0475695ccf3ae2829f/packages/editor/src/components/post-publish-panel/index.js#L146) and then [close if we don't get anything.](https://github.com/WordPress/gutenberg/blob/3da717b8d0ac7d7821fc6d0475695ccf3ae2829f/packages/editor/src/components/post-publish-panel/index.js#L50).

Reply:
Did some more testing with this. The failures drop off after around 3ms but I also managed to trigger it at 10ms so that theory is on shaky ground.

I've decided to bump this up to 100ms to match the [transition time](https://github.com/WordPress/gutenberg/blob/5c78bb872f754b52a3320ed886970dbc0583169e/packages/edit-post/src/components/layout/style.scss#L40) of the animation but I think we probably need to track the failure / retry rate of the test over time to know if we've really nailed it.

Reply:
Failure after change mentioned below now looks like this:
```
  1) [WPCOM] Likes: (desktop)
       Like posts and comments @parallel
         Login, create a new post and view it:
     TimeoutError: Waiting for the element to become located and visible By(css selector, .editor-post-publish-panel__header button[aria-label="Close panel"])
Wait timed out after 22244ms
      at /home/louis/wp/wp-calypso/node_modules/selenium-webdriver/lib/webdriver.js:842:17
      at processTicksAndRejections (internal/process/task_queues.js:93:5)
```

Reply:
Thanks for digging into this, @lsl! 🙇 I've tried another approach here - instead of _sleeps_, I'm checking if the element is still moving before clicking it (https://github.com/Automattic/wp-calypso/pull/52089/commits/7b32124a4206811202c5cadf284b038402255e21). Including that in the `clickWhenClickable` helper should fix all the issues with moving "unclickable" elements (https://github.com/Automattic/wp-calypso/pull/52089/commits/abd6dd52c4e89068ffaf5861da2a9b0577cb6ca9). I did a quick scan for places that use sleep before clicking elements and removed those _sleeps_ as they should be obsolete now (https://github.com/Automattic/wp-calypso/pull/52089/commits/29ed06261a7613b23287c69adc3d70c74e31ff77). 🤞 In my local testing, I'm able to consistently get pass with this approach.

Reply:
Nice work, much better than sprinkling sleeps everywhere.